src/nodeDef/service.ts

```diff
@@ -40,4 +40,13 @@ export interface NodeDefService {
     user: User
     surveyId: number
   }): Promise<{ [nodeDefUuid: string]: NodeDef<NodeDefType> }>
+
+
+  // ==== VIRTUAL ENTITIES
+  countVirtualEntities(options: { surveyId: number }): Promise<number>
+
+
+  getManyVirtualEntities(options: { limit?: number; offset?: number; surveyId: number }): Promise<Array<NodeDefType>>
```

Comment:
Doesn't this return an array of NodeDef not NodeDefType?

Reply:
Promise<Array<NodeDef<NodeDefType>>>